distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
 
     }
 
+    @Test
+    public void testDownloadFromFederatedCswSource() throws Exception {
+        RestitoRetrievalResponse restitoRetrievalResponse =
+                new RestitoRetrievalResponse("testData");
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"))
+                .then(getCswRetrievalHeaders(), restitoRetrievalResponse.action());
+
+        String restUrl =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                        + "?transform=resource";
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
+    }
+
+    @Test
+    public void testSlowResponse() throws Exception {
+        RestitoRetrievalResponse slowAction = new RestitoRetrievalResponse("testData", 2000);
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"))
+                .then(getCswRetrievalHeaders(), slowAction.action());
+
+        String restUrl =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                        + "?transform=resource";
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
```

Comment:
Looked into it a bit more. After not receiving anything for 6 seconds, our Future<ReliableResourceStatus> throws a cancellation exception and then waits 10 seconds before reattempting the download. I decided to remove this test altogether as it does not seem to be testing anything useful that the other tests aren't already testing.